errors/multi.go

```diff
@@ -0,0 +1,145 @@
+package errors
+
+import (
+	"fmt"
+	"strings"
+)
+
+var _ coder = (*multiErr)(nil)
+var _ causer = (*multiErr)(nil)
+var _ Multi = (*multiErr)(nil)
+
+// Multi is an interface for multiErr to avoid exposing actual implementation
+// to the rest of the app.
+type Multi interface {
```

Comment:
I think this interface is not needed. There is `MultiAdd` and `MultiAddNamed` to construct the error, that do not leak this interface. It is also not used anywhere.

Reply:
It is used in tests

Reply:
and it is also used in Validation method in batch, please take a closer look.